python/lsst/ts/ess/csc/data_client/controller_data_client.py

```diff
@@ -329,18 +294,22 @@ async def run_command(self, command: str, **parameters: Any) -> None:
             If it takes more than COMMUNICATE_TIMEOUT seconds
             to acquire the lock or write the data.
         """
-        json_str = json.dumps({"command": command, "parameters": parameters})
+        data = {
+            common.JsonKeys.COMMAND: command,
+            common.JsonKeys.PARAMETERS: parameters,
+        }
         await asyncio.wait_for(
-            self._basic_run_command(json_str), timeout=COMMUNICATE_TIMEOUT
+            self._basic_run_command(data), timeout=COMMUNICATE_TIMEOUT
         )
 
-    async def _basic_run_command(self, json_str: str) -> None:
+    async def _basic_run_command(self, data: typing.Any) -> None:
```

Comment:
Please consider using `dict[str, typing.Any]` or even just `dict` as the type hint (and in the Parameters section below).